parser/ast_nodes.js

```diff
@@ -0,0 +1,56 @@
+"use strict";
+
+const {ast_types} = require("../constants/ast");
+
+/* Tag Node */
+function tag(key) {
+    this.key = key;
+    this.attributes = [];
+    this.children = []; 
+};
+
+tag.prototype = {
+    type : ast_types.tag,
+
+    addChild : child => this.children.push(child),
```

Comment:
We can't use arrow functions as function methods. Since they are lexically scoped the value of `this` won't be the instance of the object.

Reply:
Ah yeah, completely forgot about that. Thanks for pointing that out.